tests/pems/permutations/generate-certs.sh

```diff
@@ -121,25 +138,45 @@ cert-gen () {
     echo "verifying client certificates"
     openssl verify -CAfile ca-cert.pem client-cert.pem
 
-    echo "cleaning up temporary files"
```

Comment:
One thing I just noticed is that the ca-cert gets added to the server-chain.pem (line 132). It probably shouldn't be in there, root certs aren't actually sent over the wire since the verifier should already have it.

Reply:
At least for AWS services, it seems to be common practice to include the root CA in the chain sent by the server. E.g. S3 returns the leaf, intermediate, root CA, and old CA that cross-signed the Amazon Root